testplan/runnable.py

```diff
@@ -419,10 +427,88 @@ def _create_result(self):
                     test_results[uid] = resource_result.result
             else:
                 test_results[uid] = resource_result
-            self._result.test_report.append(test_results[uid].report)
+
+            report = test_results[uid].report
+            if report.part and self.cfg.merge_scheduled_parts:
+                # Save the report temporarily and later will merge it
+                test_rep_lookup.setdefault(report.uid, []).append(
+                    (test_results[uid].run, report))
+                try:
+                    test_report.get_by_uid(report.uid)
+                except KeyError:
+                    # Will be appended to test_report as a placeholder
+                    report = report.__class__(
+                        report.name, category=report.category, uid=report.uid)
+                else:
+                    continue
+            elif report.part:
+                report.name = '{} - part({}/{})'.format(
+                    report.name, report.part[0], report.part[1])
```

Comment:
maybe should use report.part[0] + 1 here for readability, so parts are e.g. part 1/3, 2/3, 3/3.

Reply:
I forgot the old version testplan displays part 0 to N-1 or 1 to N. I'll check and make it the same,